internal/usecases/pegin/accept_pegin_quote.go

```diff
@@ -97,6 +107,69 @@ func (useCase *AcceptQuoteUseCase) Run(ctx context.Context, quoteHash string) (q
 	}, nil
 }
 
+func (useCase *AcceptQuoteUseCase) handleTrustedAccountSignature(ctx context.Context, quoteHash string, signature string, peginQuote *quote.PeginQuote) (liquidity_provider.TrustedAccountDetails, error) {
+	if signature == "" {
+		return liquidity_provider.TrustedAccountDetails{}, nil
+	}
+	trustedAccount, err := useCase.getTrustedAccount(ctx, quoteHash, useCase.lp.GetSigner(), signature)
+	if err != nil {
+		return liquidity_provider.TrustedAccountDetails{}, err
+	}
+	if err = useCase.checkLockingCap(ctx, trustedAccount, peginQuote); err != nil {
+		return liquidity_provider.TrustedAccountDetails{}, err
+	}
+	return trustedAccount, nil
+}
+
+func (useCase *AcceptQuoteUseCase) getTrustedAccount(ctx context.Context, quoteHash string, signer entities.Signer, signature string) (liquidity_provider.TrustedAccountDetails, error) {
+	address, err := usecases.RecoverSignerAddress(quoteHash, signature)
+	if err != nil {
+		return liquidity_provider.TrustedAccountDetails{}, err
+	}
+
+	trustedAccount, err := liquidity_provider.ValidateConfiguration("accept pegin quote", signer, func() (*entities.Signed[liquidity_provider.TrustedAccountDetails], error) {
+		return useCase.trustedAccountRepository.GetTrustedAccount(ctx, address)
+	})
+	if err != nil {
+		return liquidity_provider.TrustedAccountDetails{}, liquidity_provider.ErrTamperedTrustedAccount
+	}
+	return trustedAccount.Value, nil
+}
+
+func (useCase *AcceptQuoteUseCase) checkLockingCap(ctx context.Context, trustedAccount liquidity_provider.TrustedAccountDetails, peginQuote *quote.PeginQuote) error {
+	errorArgs := usecases.NewErrorArgs()
+
+	// Get all retained quotes for this trusted account
+	quotes, err := useCase.quoteRepository.GetRetainedQuotesForAddress(ctx, trustedAccount.Address)
```

Comment:
this is getting all the retained quotes, not only the pending ones, so this will count the quotes that already finished for the locking cap which is incorrect. I'd add a variadic argument to this function to pass the states I want to recover so you can get the quotes from that address in X state

Reply:
I don't think so, probably I'm missing something. Please check the implementation of the function and notice that it is filtering by status of the quote
https://github.com/rsksmart/liquidity-provider-server/blob/6a28e0b9f7afdcdb1ff774ec379e397af5bb240d/internal/adapters/dataproviders/database/mongo/pegin.go#L213

Reply:
I changed the function to receive the states.